interfaces/builtin/shared_memory.go

```diff
@@ -198,37 +228,93 @@ func writeSharedMemoryPaths(w io.Writer, slot *interfaces.ConnectedSlot,
 }
 
 func (iface *sharedMemoryInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
+	privateAttr, isSet := plug.Attrs["private"]
+	private, ok := privateAttr.(bool)
+	if isSet && !ok {
+		return fmt.Errorf(`shared-memory "private" attribute must be a bool, not %v`, privateAttr)
+	}
+	if plug.Attrs == nil {
+		plug.Attrs = make(map[string]interface{})
+	}
+	plug.Attrs["private"] = private
+
 	sharedMemoryAttr, isSet := plug.Attrs["shared-memory"]
```

Comment:
nitpick, we should rename one of the `isSet` vars so it's more clear when it's used below in the `if private {` branch which one is being referred to